bot.py

```diff
@@ -41,15 +43,49 @@ def pastebin(command):
     if response["status_txt"] == "OK":
         return "output truncated; full output at: " + response["data"]["url"]
     else:
+        print(response)
+        return "failed to shorten url"
+
+def gist(version, code):
+    url = "https://api.github.com/gists"
+
+    r = requests.post("https://api.github.com/gists",
+            json = {
+                "description": "Shared via Rust Playground",
+                "public": True,
+                "files": {
+                    "playbot.rs": {
+                        "content": code
+                    }
+                }
+            })
```

Comment:
[This would indeed be an issue](https://developer.github.com/v3/#rate-limiting).
5000 gists per hour should be fine for the IRC channel though, unless someone abuses it...